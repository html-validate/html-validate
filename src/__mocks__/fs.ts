import { fs as memfs } from "memfs";

const fs = {
	__esModule: true,
	existsSync: memfs.existsSync.bind(memfs),
	lstatSync: memfs.lstatSync.bind(memfs),
	mkdirSync: memfs.mkdirSync.bind(memfs),
	readFileSync: memfs.readFileSync.bind(memfs),
	readdir: memfs.readdir.bind(memfs),
	readdirSync: memfs.readdirSync.bind(memfs),
	readlinkSync: memfs.readlinkSync.bind(memfs),
	realpath: memfs.realpath.bind(memfs),
	realpathSync: { native: memfs.realpathSync.bind(memfs) },
	statSync: memfs.statSync.bind(memfs),
	writeFile: memfs.writeFile.bind(memfs),
	writeFileSync: memfs.writeFileSync.bind(memfs),
};

const proxy = new Proxy(fs, {
	get(target, prop, receiver): any {
		if (prop === "default") {
			return proxy;
		}
		if (Reflect.has(target, prop)) {
			return Reflect.get(target, prop, receiver);
		} else {
			throw new Error(`fs mock does not implement "${prop.toString()}"`);
		}
	},
});

module.exports = proxy;
