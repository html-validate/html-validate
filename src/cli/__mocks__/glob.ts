import minimatch from "minimatch";

const glob: any = jest.requireActual("glob");

interface Options {
	cwd?: string;
}

let mockFiles: string[] = null;

function setMockFiles(files: string[]): void {
	mockFiles = files;
}

function resetMock(): void {
	mockFiles = null;
}

const originalSync = glob.sync;
function syncMock(pattern: string, options: Options = {}): string[] {
	if (!mockFiles) {
		return originalSync(pattern, options);
	}

	/* slice the cwd away since it is prepended automatically and makes it harder
	 * to test with */
	const cwd = options.cwd || "";
	const dir = cwd.replace(process.cwd(), "").replace(/^\/(.*)/, "$1/");

	let src = mockFiles;
	if (dir) {
		src = src
			.filter(cur => cur.startsWith(dir))
			.map(cur => cur.slice(dir.length));
	}

	return src.filter(cur => minimatch(cur, pattern));
}

glob.setMockFiles = setMockFiles;
glob.resetMock = resetMock;
glob.sync = syncMock;

module.exports = glob;
