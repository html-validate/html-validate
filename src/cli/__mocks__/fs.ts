import path from "path";
import { Stats } from "fs";

interface Entry {
	type: "file" | "directory";
}

interface FileEntry extends Entry {
	type: "file";
	content: string;
}

interface DirectoryEntry extends Entry {
	type: "directory";
}

class FSError extends Error {
	public code: string;
	public path: string;

	public constructor(code: string, message: string, path: string) {
		super(message);
		this.code = code;
		this.path = path;
	}

	public toString(): string {
		return `${this.code}: ${this.message} '${this.path}'`;
	}
}

class ENOENT extends FSError {
	public constructor(filePath: string) {
		super("ENOENT", "no such file or directory", filePath);
	}
}

let mockData: Map<string, Entry> = new Map();

function isDirectory(entry: Entry): entry is DirectoryEntry {
	return entry.type === "directory";
}

function isFile(entry: Entry): entry is FileEntry {
	return entry.type === "file";
}

function normalize(filePath: string): string {
	return path.isAbsolute(filePath) ? filePath : path.join(process.cwd(), filePath);
}

function mockFile(filePath: string, content: string): void {
	const normalized = normalize(filePath);
	const fileEntry: FileEntry = {
		type: "file",
		content,
	};
	mockData.set(normalized, fileEntry);
	let current = path.dirname(normalized);
	while (current !== "/") {
		if (mockData.has(current)) {
			break;
		}
		const dirEntry: DirectoryEntry = {
			type: "directory",
		};
		mockData.set(current, dirEntry);
		current = path.dirname(current);
	}
}

function mockReset(): void {
	mockData = new Map();
	const rootEntry: DirectoryEntry = {
		type: "directory",
	};
	mockData.set("/", rootEntry);
}

function existsSync(filePath: string): boolean {
	return mockData.has(normalize(filePath));
}

function readdirSync(
	filePath: string,
	options?: string | { encoding?: string; withFileTypes?: boolean }
): string[] | Buffer[] {
	if (!filePath) {
		throw new ENOENT(filePath);
	}
	const normalized = normalize(filePath);
	const entry = mockData.get(normalized);
	if (!entry) {
		throw new ENOENT(filePath);
	}
	if (!isDirectory(entry)) {
		throw new FSError("ENOTDIR", "not a directory", filePath);
	}
	const encoding = typeof options === "string" ? options : options?.encoding ?? "utf-8";
	const entries = Array.from(mockData.keys())
		.filter((it) => path.dirname(it) === normalized)
		.map((it) => path.basename(it))
		.sort();
	if (typeof options !== "string" && options?.withFileTypes) {
		throw new Error("NotImplemented: readdirSync option withFileTypes is not supported by mock");
	}
	if (encoding !== "utf-8") {
		throw new Error("NotImplemented: readdirSync encoding only supports utf-8");
	}
	return entries;
}

function readFileSync(
	filePath: string,
	options?: string | { encoding?: string; flag?: string }
): Buffer | string {
	if (!filePath) {
		throw new ENOENT(filePath);
	}
	const normalized = normalize(filePath);
	const entry = mockData.get(normalized);
	if (!entry) {
		throw new ENOENT(filePath);
	}
	if (!isFile(entry)) {
		throw new FSError("EISDIR", "illegal operation on a directory", filePath);
	}
	if (typeof options === "string" || options?.encoding) {
		return entry.content;
	} else {
		return Buffer.from(entry.content, "utf-8");
	}
}

function statSync(filePath: string): Stats {
	if (!filePath) {
		throw new ENOENT(filePath);
	}
	const normalized = normalize(filePath);
	const date = new Date();
	const time = date.getTime();
	const entry = mockData.get(normalized);
	if (!entry) {
		throw new ENOENT(filePath);
	}
	return {
		isFile: () => isFile(entry),
		isDirectory: () => isDirectory(entry),
		isBlockDevice: () => false,
		isCharacterDevice: () => false,
		isSymbolicLink: () => false,
		isFIFO: () => false,
		isSocket: () => false,
		dev: 1,
		ino: 1,
		mode: 0b110110110,
		nlink: 1,
		uid: 1,
		gid: 1,
		rdev: 1,
		size: 1,
		blksize: 1,
		blocks: 1,
		atimeMs: time,
		mtimeMs: time,
		ctimeMs: time,
		birthtimeMs: time,
		atime: date,
		mtime: date,
		ctime: date,
		birthtime: date,
	};
}

const original = jest.requireActual("fs");
const fs = {
	__esModule: undefined,

	mockFile,
	mockReset,

	existsSync,
	realpath: original.realpath,
	realpathSync: original.realpathSync,
	readdirSync,
	readFileSync,
	statSync,
	lstatSync: statSync,
};

module.exports = new Proxy(fs, {
	get(target, prop, receiver) {
		if (Reflect.has(target, prop)) {
			return Reflect.get(target, prop, receiver);
		} else {
			throw new Error(`fs mock does not implement "${prop.toString()}"`);
		}
	},
});
