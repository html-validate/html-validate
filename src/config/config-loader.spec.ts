import fs from "fs";
import glob from "glob";
import path from "path";
import { Config, ConfigData, ConfigLoader, RuleConfig } from ".";
import HtmlValidate from "../htmlvalidate";

declare module "./config-data" {
	interface ConfigData {
		mockFilenames?: string[];
	}
}

class MockConfig {
	public static empty(): Config {
		return Config.empty();
	}

	public static fromFile(filename: string): Config {
		return Config.fromObject({
			/* set root to true for if the last directory name is literal "root" */
			root: path.basename(path.dirname(filename)) === "root",

			mockFilenames: [filename],
		});
	}
}

function getInteralCache(loader: ConfigLoader): Map<string, Config> {
	return (loader as any).cache;
}

describe("ConfigLoader", () => {
	let loader: ConfigLoader;

	afterEach(() => {
		jest.restoreAllMocks();
	});

	describe("fromTarget()", () => {
		beforeEach(() => {
			loader = new ConfigLoader(MockConfig);
		});

		it("should load configuration", () => {
			jest.spyOn(fs, "existsSync").mockImplementation((filename: string) => {
				return filename === path.resolve("/path/to/.htmlvalidate.json");
			});
			const config = loader.fromTarget("/path/to/target.html");
			expect(config.get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/path/to/.htmlvalidate.json"),
					],
				})
			);
		});

		it("should load configuration from parent directory", () => {
			jest.spyOn(fs, "existsSync").mockImplementation((filename: string) => {
				return filename === path.resolve("/path/.htmlvalidate.json");
			});
			const config = loader.fromTarget("/path/to/target.html");
			expect(config.get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/path/.htmlvalidate.json"),
					],
				})
			);
		});

		it("should load configuration from multiple files", () => {
			jest.spyOn(fs, "existsSync").mockImplementation(() => true);
			const config = loader.fromTarget("/path/to/target.html");
			expect(config.get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/.htmlvalidate.json"),
						path.resolve("/path/.htmlvalidate.json"),
						path.resolve("/path/to/.htmlvalidate.json"),
					],
				})
			);
		});

		it("should stop searching when root is found", () => {
			jest.spyOn(fs, "existsSync").mockImplementation(() => true);
			const config = loader.fromTarget("/project/root/src/target.html");
			expect(config.get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/project/root/.htmlvalidate.json"),
						path.resolve("/project/root/src/.htmlvalidate.json"),
					],
				})
			);
		});

		it("should load empty config for inline sources", () => {
			const config = loader.fromTarget("inline");
			expect(config.get()).toEqual(Config.empty().get());
		});

		it("should cache results", () => {
			jest.spyOn(fs, "existsSync").mockImplementation(() => true);
			const cache = getInteralCache(loader);
			expect(cache.has("/path/to/target.html")).toBeFalsy();
			loader.fromTarget("/path/to/target.html");
			expect(cache.has("/path/to/target.html")).toBeTruthy();
			expect(cache.get("/path/to/target.html").get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/.htmlvalidate.json"),
						path.resolve("/path/.htmlvalidate.json"),
						path.resolve("/path/to/.htmlvalidate.json"),
					],
				})
			);
		});

		it("should load from cache if present", () => {
			jest.spyOn(fs, "existsSync").mockImplementation(() => true);
			const cache = getInteralCache(loader);
			loader.fromTarget("/path/to/target.html");
			expect(cache.has("/path/to/target.html")).toBeTruthy();
			jest.spyOn(MockConfig, "fromFile").mockImplementation(() => {
				throw new Error("expected cache to be used");
			});
			const config = loader.fromTarget("/path/to/target.html");
			expect(config.get()).toEqual(
				expect.objectContaining({
					mockFilenames: [
						/* ConfigMock adds all visited filenames to this array */
						path.resolve("/.htmlvalidate.json"),
						path.resolve("/path/.htmlvalidate.json"),
						path.resolve("/path/to/.htmlvalidate.json"),
					],
				})
			);
		});
	});

	describe("flush()", () => {
		beforeEach(() => {
			loader = new ConfigLoader(MockConfig);
		});

		it("flush() should clear cache", () => {
			const cache = getInteralCache(loader);
			cache.set("foo", null);
			cache.set("bar", null);
			cache.set("baz", null);
			expect(cache.size).toEqual(3);
			loader.flush();
			expect(cache.size).toEqual(0);
		});

		it("flush() should clear single filename", () => {
			const cache = getInteralCache(loader);
			cache.set("foo", null);
			cache.set("bar", null);
			cache.set("baz", null);
			expect(cache.size).toEqual(3);
			loader.flush("foo");
			expect(cache.size).toEqual(2);
			expect(cache.has("foo")).toBeFalsy();
			expect(cache.has("bar")).toBeTruthy();
			expect(cache.has("baz")).toBeTruthy();
		});
	});

	describe("smoketest", () => {
		beforeAll(() => {
			loader = new ConfigLoader(Config);
		});

		/* extract only relevant rules from configuration to avoid bloat when new
		 * rules are added to recommended config */
		function filter(src: Config): ConfigData {
			const whitelisted = ["void", "deprecated", "element-permitted-content"];
			const data = src.get();
			data.rules = Object.keys(data.rules)
				.filter(key => whitelisted.includes(key))
				.reduce(
					(dst, key) => {
						dst[key] = data.rules[key];
						return dst;
					},
					{} as RuleConfig
				);
			return data;
		}

		const files = glob.sync("test-files/config/**/*.html");
		files.forEach((filename: string) => {
			it(filename, () => {
				const htmlvalidate = new HtmlValidate();
				const config = htmlvalidate.getConfigFor(filename);
				const report = htmlvalidate.validateFile(filename);
				expect(filter(config)).toMatchSnapshot();
				expect(report.results).toMatchSnapshot();
			});
		});
	});
});
