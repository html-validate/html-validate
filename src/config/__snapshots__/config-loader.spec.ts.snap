// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigLoader smoketest test-files/config/directive/disable.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "off",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable.html 2`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "test-files/config/directive/disable.html",
    "messages": Array [
      Object {
        "column": 3,
        "context": "i",
        "line": 1,
        "message": "<i> must not be self-closed",
        "offset": 2,
        "ruleId": "no-self-closing",
        "selector": "i",
        "severity": 2,
        "size": 2,
      },
      Object {
        "column": 3,
        "context": "i",
        "line": 8,
        "message": "<i> must not be self-closed",
        "offset": 212,
        "ruleId": "no-self-closing",
        "selector": "i:nth-child(4)",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<i/>Before disable, should trigger

<!-- [html-validate-disable no-self-closing] -->
<i/>After disable, should not trigger
<i/>After disable, should not trigger

<!-- [html-validate-enable no-self-closing] -->
<i/>Before after, should trigger
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-block.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "off",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-block.html 2`] = `
Array [
  Object {
    "errorCount": 4,
    "filePath": "test-files/config/directive/disable-block.html",
    "messages": Array [
      Object {
        "column": 3,
        "context": "i",
        "line": 1,
        "message": "<i> must not be self-closed",
        "offset": 2,
        "ruleId": "no-self-closing",
        "selector": "i",
        "severity": 2,
        "size": 2,
      },
      Object {
        "column": 4,
        "context": "i",
        "line": 3,
        "message": "<i> must not be self-closed",
        "offset": 43,
        "ruleId": "no-self-closing",
        "selector": "div > i",
        "severity": 2,
        "size": 2,
      },
      Object {
        "column": 3,
        "context": "i",
        "line": 11,
        "message": "<i> must not be self-closed",
        "offset": 335,
        "ruleId": "no-self-closing",
        "selector": "i:nth-child(3)",
        "severity": 2,
        "size": 2,
      },
      Object {
        "column": 3,
        "context": "i",
        "line": 16,
        "message": "<i> must not be self-closed",
        "offset": 439,
        "ruleId": "no-self-closing",
        "selector": "i:nth-child(5)",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<i/>Outside block, should trigger
<div>
	<i/>Inside block, before directive, should trigger
	<!-- [html-validate-disable-block no-self-closing] -->
	<i/>Inside block, after directive, shout not trigger
	<div>
		<i/>Inside block, after directive, shout not trigger
	</div>
	<i/>Inside block, after directive, shout not trigger
</div>
<i/>Outside block, should trigger

<div>
	<!-- [html-validate-disable-block no-self-closing] -->
</div>
<i/>Outside block, should trigger

<!-- [html-validate-disable-block no-self-closing] -->
Should handle when root element is parent but no children
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-multiple.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "off",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-multiple.html 2`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "test-files/config/directive/disable-multiple.html",
    "messages": Array [
      Object {
        "column": 2,
        "context": undefined,
        "line": 1,
        "message": "<blink> is deprecated",
        "offset": 1,
        "ruleId": "deprecated",
        "selector": "blink",
        "severity": 2,
        "size": 5,
      },
      Object {
        "column": 7,
        "context": "blink",
        "line": 1,
        "message": "<blink> must not be self-closed",
        "offset": 6,
        "ruleId": "no-self-closing",
        "selector": "blink",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<blink/>
<div>
	<!-- [html-validate-disable-block no-self-closing, deprecated] -->
	<blink/>
</div>
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-next.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "off",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/directive/disable-next.html 2`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "test-files/config/directive/disable-next.html",
    "messages": Array [
      Object {
        "column": 3,
        "context": "i",
        "line": 1,
        "message": "<i> must not be self-closed",
        "offset": 2,
        "ruleId": "no-self-closing",
        "selector": "i",
        "severity": 2,
        "size": 2,
      },
      Object {
        "column": 3,
        "context": "i",
        "line": 5,
        "message": "<i> must not be self-closed",
        "offset": 136,
        "ruleId": "no-self-closing",
        "selector": "i:nth-child(3)",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<i/>Before disable, should trigger

<!-- [html-validate-disable-next no-self-closing] -->
<i/>First after disable, should not trigger
<i/>Second after disable, should trigger
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/elements/file.html 1`] = `
Object {
  "elements": Array [
    "html5",
    "<rootDir>/test-files/config/elements/elements.json",
  ],
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "error",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/elements/file.html 2`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "test-files/config/elements/file.html",
    "messages": Array [
      Object {
        "column": 3,
        "context": undefined,
        "line": 2,
        "message": "<deprecated> is deprecated",
        "offset": 8,
        "ruleId": "deprecated",
        "selector": "div > deprecated",
        "severity": 2,
        "size": 10,
      },
      Object {
        "column": 4,
        "context": undefined,
        "line": 5,
        "message": "Element <li> is not permitted as content in <ul>",
        "offset": 67,
        "ruleId": "element-permitted-content",
        "selector": "div > ul > li",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<div>
	<deprecated>new (deprecated) element</deprecated>

	<ul>
		<li>overridden so it is disallowed</li>
		<p>overridden so it is allowed</p>
	</ul>
</div>
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/file.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "error",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/file.html 2`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "test-files/config/file.html",
    "messages": Array [
      Object {
        "column": 6,
        "context": "br",
        "line": 2,
        "message": "End tag for <br> must be omitted",
        "offset": 49,
        "ruleId": "void-content",
        "selector": null,
        "severity": 2,
        "size": 3,
      },
    ],
    "source": "<!-- default configuration yields error -->
<br></br>
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/off/error/file.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "error",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/off/error/file.html 2`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "test-files/config/off/error/file.html",
    "messages": Array [
      Object {
        "column": 6,
        "context": "br",
        "line": 2,
        "message": "End tag for <br> must be omitted",
        "offset": 42,
        "ruleId": "void-content",
        "selector": null,
        "severity": 2,
        "size": 3,
      },
    ],
    "source": "<!-- reconfigured to yield error -->
<br></br>
",
    "warningCount": 0,
  },
]
`;

exports[`ConfigLoader smoketest test-files/config/off/file.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "error",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/off/file.html 2`] = `Array []`;

exports[`ConfigLoader smoketest test-files/config/warn/file.html 1`] = `
Object {
  "extends": Array [
    "html-validate:recommended",
  ],
  "plugins": Array [],
  "root": true,
  "rules": Object {
    "deprecated": "error",
    "element-permitted-content": "error",
    "no-self-closing": "error",
  },
  "transform": Object {},
}
`;

exports[`ConfigLoader smoketest test-files/config/warn/file.html 2`] = `
Array [
  Object {
    "errorCount": 0,
    "filePath": "test-files/config/warn/file.html",
    "messages": Array [
      Object {
        "column": 6,
        "context": "br",
        "line": 2,
        "message": "End tag for <br> must be omitted",
        "offset": 44,
        "ruleId": "void-content",
        "selector": null,
        "severity": 1,
        "size": 3,
      },
    ],
    "source": "<!-- configured to yield a warning -->
<br></br>
",
    "warningCount": 1,
  },
]
`;
